src/hooks/useLocalizedConstants.ts

```diff
@@ -27,16 +28,27 @@ export const useLocalizedConstants = () => {
   );
 
   const getLocalizedPhase = useCallback(
-    (phaseEnum: PhaseEnum | null | undefined): string => {
+    (phaseEnum: PhaseEnum | string | null | undefined): string => {
       return (
         constants?.phases?.find((phase) => phase.id === phaseEnum)?.name || ''
       );
     },
     [constants],
   );
 
+  const getLocalizedPledgeFrequency = useCallback(
+    (freqEnum: PledgeFrequencyEnum | string | null | undefined): string => {
+      return (
+        constants?.pledgeFrequency?.find((freq) => freq.id === freqEnum)
```

Comment:
I found out why this isn't working. I think it's because the freqEnum you get from the contact is not the enum but the value in uppercase.

Reply:
![Screenshot 2024-10-30 at 11 30 50 AM](https://github.com/user-attachments/assets/a80c2fec-6198-4d06-a6af-e24f25b539d7)

Reply:
On the coaching page, the frequency you get is in English and you are trying to compare it to the German ENUM.

![Screenshot 2024-10-30 at 11 52 09 AM](https://github.com/user-attachments/assets/a4b6447d-f071-445d-b0e8-9cb838bde420)

See constants in German: Should the Ids of the enums change?
![Screenshot 2024-10-30 at 11 53 14 AM](https://github.com/user-attachments/assets/40ba9d32-8b5a-4e4b-b8cd-2615e82abb12)

Reply:
@caleballdrin This is the largest issue I see. Fixing this will solve a lot of the others.

Reply:
Ah yeah that is the issue. The API is translating the enum...